### PR TITLE
feat(freezeV2): add query api to get delegate max resource size

### PR DESCRIPTION
**Pull request by chaosuper12321**
**What does this PR do?**
add query api to get delegate max resource size

**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: actuator/src/main/java/org/tron/core/utils/TransactionUtil.java [hunk 1]
```diff
@@ -264,4 +269,48 @@ public TransactionSignWeight getTransactionSignWeight(Transaction trx) {
...
+    long bytesSize = 0L;
+
+    if (chainBaseManager.getDynamicPropertiesStore().supportVM()) {
+      bytesSize = transactionCapsule.getInstance().toBuilder().clearRet().build().getSerializedSize();
+    } else {
+      bytesSize = transactionCapsule.getSerializedSize();
+    }
+
+    List<Transaction.Contract> contracts = transactionCapsule.getInstance().getRawData().getContractList();
+    for (Transaction.Contract contract : contracts) {
```
Comment (lxcmyf) [inline actuator/src/main/java/org/tron/core/utils/TransactionUtil.java:284]: Can be optimized with lamda expression

File: framework/src/main/java/org/tron/core/Wallet.java [hunk 2]
```diff
@@ -826,6 +843,46 @@ public GrpcAPI.GetAvailableUnfreezeCountResponseMessage getAvailableUnfreezeCoun
     return builder.build();
   }
 
+  public long calcCanDelegatedBandWidthMaxSize(
+          ByteString ownerAddress) {
+    AccountStore accountStore = chainBaseManager.getAccountStore();
+    DynamicPropertiesStore dynamicStore = chainBaseManager.getDynamicPropertiesStore();
+    AccountCapsule ownerCapsule = accountStore.get(ownerAddress.toByteArray());
```
Comment (lxcmyf) [inline framework/src/main/java/org/tron/core/Wallet.java:850]: OwnerCapsule determines whether it is empty

File: framework/src/main/java/org/tron/core/Wallet.java [hunk 3]
```diff
@@ -826,6 +843,46 @@ public GrpcAPI.GetAvailableUnfreezeCountResponseMessage getAvailableUnfreezeCoun
...
+    long ownerNetUsage = (long) (netUsage * ((double)(ownerCapsule.getFrozenV2BalanceForBandwidth())
+            / ownerCapsule.getAllFrozenBalanceForBandwidth()));
+    long remain = ownerCapsule.getFrozenV2BalanceForBandwidth() - ownerNetUsage;
+    return  remain > 0 ? remain : 0;
+  }
+
+  public long calcCanDelegatedEnergyMaxSize(ByteString ownerAddress) {
+    AccountStore accountStore = chainBaseManager.getAccountStore();
+    DynamicPropertiesStore dynamicStore = chainBaseManager.getDynamicPropertiesStore();
+    AccountCapsule ownerCapsule = accountStore.get(ownerAddress.toByteArray());
```
Comment (lxcmyf) [inline framework/src/main/java/org/tron/core/Wallet.java:871]: OwnerCapsule determines whether it is empty

File: actuator/src/main/java/org/tron/core/utils/TransactionUtil.java [hunk 4]
```diff
@@ -264,4 +269,48 @@ public TransactionSignWeight getTransactionSignWeight(Transaction trx) {
...
+    TransactionCapsule fakeTransactionCapsule = new TransactionCapsule(builder.build()
+            , ContractType.DelegateResourceContract);
+    long size1 = consumeBandWidthSize(fakeTransactionCapsule, chainBaseManager);
+
+    DelegateResourceContract.Builder builder2 = DelegateResourceContract.newBuilder()
+                    .setBalance(TRX_PRECISION);
+    TransactionCapsule fakeTransactionCapsule2 = new TransactionCapsule(builder2.build()
+            , ContractType.DelegateResourceContract);
+    long size2 = consumeBandWidthSize(fakeTransactionCapsule2, chainBaseManager);
+    long addSize = (size1 - size2) > 0 ? (size1 - size2) : 0L;
```
Comment (lxcmyf) [inline actuator/src/main/java/org/tron/core/utils/TransactionUtil.java:311]: use Math.max() instead

File: actuator/src/main/java/org/tron/core/utils/TransactionUtil.java [hunk 5]
```diff
@@ -264,4 +269,48 @@ public TransactionSignWeight getTransactionSignWeight(Transaction trx) {
     return tswBuilder.build();
   }
 
+  public static long consumeBandWidthSize(
+          final TransactionCapsule transactionCapsule,
+          ChainBaseManager chainBaseManager) {
+    long bytesSize = 0L;
```
Comment (lxcmyf) [inline actuator/src/main/java/org/tron/core/utils/TransactionUtil.java:275]: Remove this useless assignment to local variable "bytesSize".

File: actuator/src/main/java/org/tron/core/utils/TransactionUtil.java (hunk 1 above)
Comment (chaosuper12321) [reply actuator/src/main/java/org/tron/core/utils/TransactionUtil.java:284]: contracts size is only one

File: actuator/src/main/java/org/tron/core/utils/TransactionUtil.java [hunk 6]
```diff
@@ -264,4 +269,48 @@ public TransactionSignWeight getTransactionSignWeight(Transaction trx) {
...
+    } else {
+      bytesSize = transactionCapsule.getSerializedSize();
+    }
+
+    List<Transaction.Contract> contracts = transactionCapsule.getInstance().getRawData().getContractList();
+    for (Transaction.Contract contract : contracts) {
+      if (contract.getType() == Contract.ContractType.ShieldedTransferContract) {
+        continue;
+      }
+      if (chainBaseManager.getDynamicPropertiesStore().supportVM()) {
```
[the remark this answers is not in the document]
Comment (chaosuper12321) [reply actuator/src/main/java/org/tron/core/utils/TransactionUtil.java:288]: contracts size is only one

[the remark this answers is not in the document]
Comment (chaosuper12321) [reply actuator/src/main/java/org/tron/core/utils/TransactionUtil.java]: ok

**Pull request merged**